src/hotspot/share/c1/c1_LIRGenerator.cpp

```diff
@@ -1595,13 +1595,84 @@ class TempResolvedAddress: public Instruction {
   virtual const char* name() const  { return "TempResolvedAddress"; }
 };
 
-void LIRGenerator::access_flattened_array(bool is_load, LIRItem& array, LIRItem& index, LIRItem& obj_item) {
+void LIRGenerator::access_sub_element(LIRItem& array, LIRItem& index, LIR_Opr& result, ciField* field, int sub_offset) {
```

Comment:
There is quite some code duplication here with access_flattened_array, can we factor it out?

Reply:
Code has been refactored, and the common code in access_flattened_array() and access_sub_element() is now shared in the new method get_and_load_element_address().